stistools/splice.py

```diff
@@ -0,0 +1,842 @@
+                                  overlap_sections[i]['wavelength'],
+                                  overlap_sections[i]['flux']))
+        err_interp.append(np.interp(overlap_ref['wavelength'],
+                                    overlap_sections[i]['wavelength'],
+                                    overlap_sections[i]['uncertainty']))
+        net_interp.append(np.interp(overlap_ref['wavelength'],
+                                    overlap_sections[i]['wavelength'],
+                                    overlap_sections[i]['net']))
+    f_interp = np.array(f_interp)
+    err_interp = np.array(err_interp)
+    net_interp = np.array(net_interp)
+    sens_interp = f_interp / net_interp  # This is a good estimate of the
+    # sensitivity
+    sens_ref = overlap_ref['flux'] / overlap_ref['net']
+
+    # Merge the spectra. We will take the weighted averaged, with weights equal
+    # to the inverse of the uncertainties squared multiplied by a scale factor
+    # to avoid numerical overflows.
+    if weight == 'sensitivity':
+        scale = 1E-10
+        weights_interp = sens_interp / scale
+        weights_ref = sens_ref / scale
+    elif weight == 'snr':
+        scale = 1E-20
+        weights_interp = (1 / err_interp) ** 2 * scale
+        weights_ref = (1 / overlap_ref['uncertainty']) ** 2 * scale
+    else:
+        raise ValueError(
+            'The weighting option "{}" is not implemented.'.format(weight))
+
+    # Here we deal with the data-quality flags. We only accept flags that are
+    # listed in `acceptable_dq_flags`. Let's initialize the dq flag arrays
+    dq_ref = overlap_ref['data_quality']
+
+    # We create a new data-quality array filled with 32768, which is what we
+    # establish as the flag for co-added pixels
+    dq_merge = np.ones_like(dq_ref, dtype=int) * 32768
+
+    # The interpolated dq flag array is a bit more involved. First we
+    # interpolate the original array to the new wavelength grid, and then we
+    # round all the values to the nearest integer
+    dq_interp = []
+    for i in range(n_overlaps - 1):
+        dq_interp.append(np.rint(np.interp(overlap_ref['wavelength'],
+                                           overlap_sections[i]['wavelength'],
+                                           overlap_sections[i]['data_quality']))
+                         )
+    dq_interp = np.array(dq_interp)
+    # However this does not guarantee the interpolated and rounded dq values are
+    # valid dq flags. Since the interpolation occurs at very small wavelength
+    # shifts, for now we assume that all dq flags will be valid. This may be
+    # changed in the future.
+    # We start assuming that all the dq weights are zero
+    dq_weights_ref = np.zeros_like(dq_ref)
+    dq_weights_interp = np.zeros_like(dq_interp)
+    # And then for each acceptable dq, if the element of the dq array is one
+    # of the acceptable flags, we set its dq weight to one
+    for adq in acceptable_dq_flags:
```

Comment:
Should probably check bitwise, i.e.:
```python
np.where(dq_ref & adq)
```

and
```python
np.where(dq_interp & adq)
```

In case more than one flag is present at a particular location.  Perhaps this doesn't occur with the default flags, but could with other options provided by the user.

Reply:
In fact, you can probably skip the loop over the various adq values and compare to the scalar value that is the bitwise-or or the adq flags (sum in this case).

Reply:
Ok, so wrapping it up together:

```python
from functools import reduce

acceptable_dq_flags = reduce(np.bitwise_or, acceptable_dq_flags)  # scalar value

dq_weights_ref[np.where(dq_ref & acceptable_dq_flags)] = 1
dq_weights_interp[np.where(dq_interp & acceptable_dq_flags)] = 1
```

(Should be tested.)

Reply:
It seems that this code snippet catches all the good flags, except for `0`.

Reply:
Good catch!  We don't typically flag acceptable pixels, so that slipped by.

Perhaps:
```python
bad_dq_flags = ~reduce(np.bitwise_or, acceptable_dq_flags)  # scalar value

dq_weights_ref = np.ones(..., dtype=...)
dq_weights_interp = np.ones(..., dtype=...)

dq_weights_ref[dq_ref & bad_dq_flags] = 0
dq_weights_interp[dq_interp & bad_dq_flags] = 0
```

Looking at it now, we probably don't need the `np.where` either.